src/Core/Manifest/ModuleResource.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace SilverStripe\Core\Manifest;
+
+use InvalidArgumentException;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * Experimental API; This may change at any time
+ */
+class ModuleResource
```

Comment:
Describe what this API does anyway? :D

Reply:
Since resources became more complex than a simple path, I've shifted it to an object that can be physically passed around independently of the module itself.

I'll add a class description.